lib/dash/segment_template.js

```diff
@@ -106,7 +106,7 @@ shaka.dash.SegmentTemplate = class {
       // Unless that live content is multi-period; it's safe to fit every period
       // but the last one, since only the last period might receive new
       // segments.
-      const shouldFit = periodEnd != Infinity;
+      const shouldFit = periodEnd != Infinity && !context.dynamic;
```

Comment:
Yeah, now that I look again, I think this isn't correct.  For multi-period live, we need to fit segments in all but the last period.  See the comment above.

Reply:
Yes, I saw the regression with DAI contents

Reply:
How about : 

```
      // Unless that live content is multi-period; it's safe to fit every period
      // but the last one, since only the last period might receive new
      // segments.
-    const shouldFit = periodEnd != Infinity;
+    const shouldFit = context.periodInfo.isLastPeriod? !context.dynamic : periodEnd!=Infinity;
```

if (not last period) {behave as currently}
if (last period) {fit if not static}

This should only change the behaviour for a 'last period', 'dynamic' schedule with 'defined duration' - i.e. the specific case in #4913 (and exactly as stated in the comment)?